Source/WebCore/dom/DeviceMotionController.cpp

```diff
@@ -69,7 +69,7 @@ void DeviceMotionController::didChangeDeviceMotion(DeviceMotionData* deviceMotio
 
 DeviceMotionClient& DeviceMotionController::deviceMotionClient()
 {
-    return static_cast<DeviceMotionClient&>(m_client.get());
+    return downcast<DeviceMotionClient>(m_client.get());
```

Comment:
I'll be surprised if this builds without adding `SPECIALIZE_TYPE_TRAITS_BEGIN / SPECIALIZE_TYPE_TRAITS_END` for these types.

Reply:
We should totally do this though, we just need to add the right type traits.

Reply:
There should be nothing special about DeviceMotionController. Hard for me to help without knowing what you tried and what error you got.